activity_browser/logger.py

```diff
@@ -1,185 +1,376 @@
-import inspect
-import logging
 import os
+import sys
 import time
+import inspect
+import logging
+import threading
 import appdirs
 
+from traceback import extract_tb
+from io import StringIO
+from types import TracebackType
+from typing import TextIO, Type
+
+WHITELIST = ["activity_browser", "brightway2", "bw2data", "bw2io", "C/C++"]
+EXTENDED_CONSOLE = os.environ.get("AB_EXTENDED_CONSOLE", False)
+SIMPLE_CONSOLE = os.environ.get("AB_SIMPLE_CONSOLE", False)
+
 
-class ABHandler(object):
+class LowLevelStdIO:
     """
-    Creates two logging streams, one to the console (with >=INFORMATION logs)
-    the second to a file specified with the name argument (with >=DEBUG logs)
-    and attaches these to a logger created in the calling modules.
-
-    There are 4 available levels in AB, increasing in importance:
-    - DEBUG
-    - INFO
-    - WARNING
-    - ERROR
-    - CRITICAL is available in logging but not implemented in AB
-    See also: https://docs.python.org/3/library/logging.html#levels
-
-    Provides the formats and initialization procedures for the logging streams.
+    This class will replace the filehandles of the supplied StdIO with own ones so changes to said stream can be caught.
+    It will instantiate a capturing thread that will read lines from the stream and log them using a logger called
+    "C/C++". The original StdIO filehandle is preserved so we can still write to it using the "write" method.
```

Comment:
I want the drugs you were taking when you came up with this. Is it really the best default logger name? Can we not at least namespace it?

Reply:
I'd love to share them with you one day.

But seriously, this is currently the only way to capture any QT logs as they are written directly to the stderr & stdout filehandles instead of through the Python implementation of them. 

I'm impartial on the "C/C++" name if that's what you mean. It's mostly a clear indication that the stdio write was made from outside of Python.

Reply:
As BW and the whole ecosystem becomes better developed, more and more logs will be created by libraries and apps. So my ~humble~ strong opinion is that the log names should clearly identify their origin, like "AB.layouts". Just "C/C++" might work for you for now, but "AB.pyside" is better looking forward.

Reply:
Yeah I agree on you with that. Issue is that we cannot be sure that the write was actually made by Pyside. Any library writing directly to the io filehandles will end up under this logger, they are not identifiable anymore at that point.

Sure, right now I know only of PySide that's doing this. And when we move to a newer version this point is moot anyway, because PySide6 implements Python logging. But still it would be masking what the code is doing in truth: catching all C stdio writes.